da/celestia/celestia.go

```diff
@@ -111,6 +94,33 @@ func (c *DataAvailabilityLayerClient) Init(config []byte, pubsubServer *pubsub.S
 	return nil
 }
 
+func createConfig(bz []byte) (c Config, err error) {
+	if len(bz) <= 0 {
+		return c, errors.New("supplied config is empty")
+	}
+	err = json.Unmarshal(bz, &c)
+	if err != nil {
+		return c, fmt.Errorf("json unmarshal: %w", err)
+	}
+
+	err = c.InitNamespaceID()
+	if err != nil {
+		return c, fmt.Errorf("init namespace id: %w", err)
+	}
+
+	if c.GasPrices == 0 {
+		return c, errors.New("gas prices must be set")
+	}
+
+	if c.RetryDelay == 0 {
```

Comment:
missing default for `RetryAttempts`

Reply:
its not missing, 0 is a valid value so you cant disambiguate

Reply:
why not setting `defaultRetryAttempts` to be `0`?
I think it's more explicit